sporkfish/searcher/move_ordering.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABC, abstractmethod
+
+import chess
+
+from ..board.board import Board
+
+
+class MoveOrder(ABC):
+    @abstractmethod
+    def evaluate(self, board: Board, move: chess.Move) -> float:
+        pass
+
+
+class MvvLvaHeuristic(MoveOrder):
+    def evaluate(self, board: Board, move: chess.Move) -> float:
+        """
+        Calculate the Most Valuable Victim - Least Valuable Aggressor heuristic value
+        for a capturing move based on the value of the captured piece.
+
+        Parameters:
+            board (Board): The chess board.
+            move (chess.Move): The capturing move.
+
+        Returns:
+            int: The heuristic value of the capturing move based on the value of the captured piece.
+        """
+
+        # make this configurable in future (less error prone)
```

Comment:
Why make this configurable?

Reply:
At the moment, the `MVV_LVA` matrix is hard coded in the function:

```
# make this configurable in future (less error prone)
        # Columns: attacker P, N, B, R, Q, K
        MVV_LVA = [
            [15, 14, 13, 12, 11, 10],  # victim P
            [25, 24, 23, 22, 21, 10],  # victim N
            [35, 34, 33, 32, 31, 30],  # victim B
            [45, 44, 43, 42, 41, 40],  # victim R
            [55, 54, 53, 52, 51, 50],  # victim Q
            [0, 0, 0, 0, 0, 0],  # victim K
        ]
```
There are a couple of reasons I don't quite like hard coded numbers (in general, not just about this matrix):

- Seems a bit like a black box, no one really understands the behaviour and heuristics of these numbers - and it is hard to test
- Makes it a bit hard to unit test and maintain:
	- The way the MVV LVA unit tests works is by setting up simple board that allows for a small set of capturing legal moves, and try to replicate the numbers in the test. Since this matrix is not accessible outside of the scope of the function, I had to hard code the numbers to match the values. 
	- If (or when) someone does change something in this matrix, either the test will break, and the developer would need to fix the test manually, or worse, the change is not tested.

Hence, my suggestion is either (a) make this matrix configurable, or (b) make this matrix a member (@property) within the class, so at least it is accessible by clients.

Of course, this is not high priority, and suspect no plans to play around with the numbers in this matrix. 

LMKYT?

Reply:
I'd argue these values deliberately shouldn't be configurable. Essentially these values are chosen such that so that a) they're all different, to ensure when you evaluate a move order, you always get the same order, and b)

Reply:
Is there a typo in the matrix?

```
        # make this configurable in future (less error prone)
        # Columns: attacker P, N, B, R, Q, K
        MVV_LVA = [
            [15, 14, 13, 12, 11, 10],  # victim P
            [25, 24, 23, 22, 21, 10],  # victim N
            [35, 34, 33, 32, 31, 30],  # victim B
            [45, 44, 43, 42, 41, 40],  # victim R
            [55, 54, 53, 52, 51, 50],  # victim Q
            [0, 0, 0, 0, 0, 0],  # victim K
        ]
```

See `MVV_LVA[1][5]` (K takes N) - suspect it should be 20?

Reply:
Yes - if you don't mind fixing it for me pls :)

Reply:
Sure beans - pleasure innit.

The test idea is interesting. But would prefer add in a different PR (wanna keep PRs small avoid massive conflicts...!)

Reply:
fixed - also put the matrix a member variable of the class

Reply:
Ok, thanks!

Reply:
Could we remove the comment though then please?